salt/modules/grains.py

```diff
@@ -289,8 +289,8 @@ def setvals(grains, destructive=False, refresh=True):
         msg = 'Unable to write to cache file {0}. Check permissions.'
         log.error(msg.format(fn_))
     if not __opts__.get('local', False):
-        # Sync the grains
-        __salt__['saltutil.sync_grains'](refresh=refresh)
```

Comment:
This change from `sync_grains` to `refresh_grains` right here is causing all of the test failures in this merge forward. I think these tests might be showing a change in behavior.

@terminalmage since this was your change that is getting merged forward and you're more familiar with the reasoning behind this move, would you mind taking a look when you get a chance? The related PR is #39619.